app/src/main/java/com/example/admin/showdata/DownloadData.java

```diff
@@ -0,0 +1,72 @@
+package com.example.admin.showdata;
+
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by admin on 15-02-2018.
+ */
+
+public class DownloadData {
+    HttpURLConnection connection;
+    List<ContentModel> data_list;
+
+
+    public List<ContentModel> getdatafromurl(String passed_url){
+        StringBuilder result = new StringBuilder();
+        try {
+            // getting url
+            URL url = new URL(passed_url);
+            // setting connection
+            connection = (HttpURLConnection) url.openConnection();
+            // setting stream
+            InputStream in = new BufferedInputStream(connection.getInputStream());
```

Comment:
`InputStream` and `BufferedReader` are not closed after its use. This causes a leak of resources used by them. Use [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) for both. They close automatically when your done with them.